crates/networking/p2p/utils.rs

```diff
@@ -64,6 +72,56 @@ pub fn get_account_storages_snapshot_file(directory: &Path, chunk_index: u64) ->
     directory.join(format!("account_storages_chunk.rlp.{chunk_index}"))
 }
 
+#[cfg(feature = "rocksdb")]
+pub fn dump_accounts_to_rocks_db(
+    path: &Path,
+    mut contents: Vec<(H256, AccountState)>,
+) -> Result<(), rocksdb::Error> {
+    contents.sort_by_key(|(k, _)| *k);
+    contents.dedup_by_key(|(k, _)| {
+        let mut buf = [0u8; 32];
+        buf[..32].copy_from_slice(&k.0);
+        buf
+    });
+    let mut buffer: Vec<u8> = Vec::new();
+    let writer_options = rocksdb::Options::default();
+    let mut writer = rocksdb::SstFileWriter::create(&writer_options);
+    writer.open(std::path::Path::new(&path))?;
+    for (key, acccount) in contents {
+        buffer.clear();
+        acccount.encode(&mut buffer);
+        writer.put(key.0.as_ref(), buffer.as_slice())?;
+    }
+    writer.finish()
+}
+
+#[cfg(feature = "rocksdb")]
+pub fn dump_storages_to_rocks_db(
+    path: &Path,
+    mut contents: Vec<(H256, H256, U256)>,
+) -> Result<(), rocksdb::Error> {
+    contents.sort();
+    contents.dedup_by_key(|(k0, k1, _)| {
+        let mut buffer = [0_u8; 64];
+        buffer[0..32].copy_from_slice(&k0.0);
+        buffer[32..64].copy_from_slice(&k1.0);
+        buffer
+    });
```

Comment:
Not necessary to fix now, but I believe the following should work and be more brief:
```suggestion
    contents.dedup_by_key(|(k0, k1, _)| (k0, k1));
```